app/controllers/RequiredGiinController.scala

```diff
@@ -46,25 +47,34 @@ class RequiredGiinController @Inject() (
 
   val form: Form[String] = formProvider()
 
-  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData) {
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen RequireData) {
     implicit request =>
-      val preparedForm = request.userAnswers.flatMap(_.get(RequiredGiinPage)) match {
+      val preparedForm = request.userAnswers.get(RequiredGiinPage) match {
```

Comment:
it would be better to add this logic inside ValidXMLPage check

Reply:
Ooh i see what you mean - yeah good point ill explore that